File: app/src/main/java/com/sewageproject/base/BaseFragment.java
package com.sewageproject.base;
import com.trello.rxlifecycle2.components.support.RxFragment;
public  abstract class BaseFragment extends RxFragment {
    //当前类是否处理ui逻辑
    protected final int disableLayout = -10086;
    private boolean isInitView = false;
    private boolean isVisible = false;
    @Override
    public void setUserVisibleHint(boolean isVisibleToUser) {
        super.setUserVisibleHint(isVisibleToUser);

        //isVisibleToUser这个boolean值表示:该Fragment的UI 用户是否可见，获取该标志记录下来
        if (isVisibleToUser) {
            isVisible = true;
            isCanLoadData();
        } else {
            isVisible = false;
        }
    }
    private void isCanLoadData() {
        //所以条件是view初始化完成并且对用户可见
        if (isInitView && isVisible) {
            onLazyLoad();
            //防止重复加载数据
            isInitView = false;
            isVisible = false;
        }
    }
    /**
     * 加载要显示的数据
     */
    protected void onLazyLoad() {

    }
}

File: app/src/main/java/com/sewageproject/base/BaseFragment.kt
package com.sewageproject.base

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.databinding.ViewDataBinding
import androidx.fragment.app.Fragment
import com.gyf.immersionbar.ImmersionBar
import com.sewageproject.R
import kotlinx.coroutines.CoroutineScope
import org.greenrobot.eventbus.EventBus
import org.greenrobot.eventbus.Subscribe

/**
 *Created by wanghai
 *Date : 2020/12/23
 *Describe :
 */
abstract class BaseFragment<V : ViewDataBinding>: Fragment() {
    protected abstract fun getResourceId(): Int
    protected abstract fun initView()
    protected abstract fun initData()
    protected abstract fun initListener()
    protected open fun statusBarDark(): Boolean {
        return true
    }

    var binding: V? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        binding = DataBindingUtil.inflate(inflater, getResourceId(), null, false)
        ImmersionBar.with(this).statusBarDarkFont(statusBarDark()).keyboardEnable(true)
            .navigationBarColor(R.color.white).init()
        EventBus.getDefault().register(this)
        return binding?.root
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        initView()
        initListener()
        initData()
    }

    override fun onDestroyView() {
        if (EventBus.getDefault().isRegistered(this)) {
            EventBus.getDefault().unregister(this)
        }
        binding = null
        super.onDestroyView()
    }
    @Subscribe
    open fun fragmenEvent(event: MainEventBean?) {
    }
}